bpf/headers/go_context.h

```diff
@@ -0,0 +1,91 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _GO_CONTEXT_H_
+#define _GO_CONTEXT_H_
+
+#include "bpf_helpers.h"
+
+#define MAX_DISTANCE 10
+
+struct
+{
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, void *);
+    __type(value, struct span_context);
+    __uint(max_entries, MAX_CONCURRENT_SPANS);
+    __uint(pinning, LIBBPF_PIN_BY_NAME);
+} tracked_spans SEC(".maps");
+
+struct
+{
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, struct span_context);
+    __type(value, void *);
+    __uint(max_entries, MAX_CONCURRENT_SPANS);
+    __uint(pinning, LIBBPF_PIN_BY_NAME);
+} tracked_spans_by_sc SEC(".maps");
+
+static __always_inline void *get_parent_go_context(void *ctx, void *map) {
```

Comment:
Taken as-is from Otel, I was wondering if we should convert to use Beyla's parent finding, e.g. find_parent_goroutine()